cpp/src/utilities/default_stream.cpp

```diff
@@ -25,7 +25,7 @@ namespace cudf {
  */
 bool is_ptds_enabled()
 {
-#ifdef CUDA_API_PER_THREAD_DEFAULT_STREAM
+#if defined(CUDF_USE_PER_THREAD_DEFAULT_STREAM)
```

Comment:
I didn't know this API existed. Where is it being used?

It is worth some thought of what the intent of this API is. There are two kinds of PTDS we're dealing with now:
1. `CUDA_API_PER_THREAD_DEFAULT_STREAM` which carries the usual CUDA PTDS semantics of applying only to the library compiled with it
2. `CUDF_USE_PER_THREAD_DEFAULT_STREAM` which is a superset of 1. and carries the additional semantics of passing the PTDS to external library calls.

Which one is of interest to this function? The answer depends on how/where it's used.

Reply:
It turns out that this is primarily there so JNI code can check to ensure that it is running in the same mode as CUDF:
```
  // make sure libcudf and the JNI library are built with the same PTDS mode
  if (cudf::is_ptds_enabled() != cudf::jni::is_ptds_enabled) {
    std::ostringstream ss;
    ss << "Libcudf is_ptds_enabled=" << cudf::is_ptds_enabled()
       << ", which does not match cudf jni is_ptds_enabled=" << cudf::jni::is_ptds_enabled
       << ". They need to be built with the same per-thread default stream flag.";
    env->ThrowNew(env->FindClass("java/lang/RuntimeException"), ss.str().c_str());
    return JNI_ERR;
  }

```
Given this, and that @jlowe suggested we use `CUDA_API_PER_THREAD_DEFAULT_STREAM` in the JNI version, I think this should be ifdef'd `CUDA_API_PER_THREAD_DEFAULT_STREAM`.